packages/firebase_ml_custom/example/lib/main.dart

```diff
@@ -29,61 +29,98 @@ class _MyAppState extends State<MyApp> {
   final picker = ImagePicker();
   File _image;
   List<Map<dynamic, dynamic>> _labels;
+  //When the model is ready, _loaded changes to trigger the screen state change.
   Future<String> _loaded = loadModel();
 
+  // Triggers selection of an image and the consequent inference
   Future<void> getImageLabels() async {
-    final pickedFile = await picker.getImage(source: ImageSource.gallery);
-    final image = File(pickedFile.path);
-    if (image == null) {
-      return;
+    try {
+      final pickedFile = await picker.getImage(source: ImageSource.gallery);
+      final image = File(pickedFile.path);
+      if (image == null) {
+        return;
+      }
+      var labels = List<Map>.from(await Tflite.runModelOnImage(
+        path: image.path,
+        imageStd: 127.5,
+      ));
+      setState(() {
+        _labels = labels;
+        _image = image;
+      });
+    }
+    catch (exception) {
+      print("Failed on getting your image and it's labels: $exception");
+      print('Continuing with the program...');
+      rethrow;
     }
-    var labels = List<Map>.from(await Tflite.runModelOnImage(
-      path: image.path,
-      imageStd: 127.5,
-    ));
-    setState(() {
-      _labels = labels;
-      _image = image;
-    });
   }
 
   static Future<String> loadModel() async {
-    var modelFile = await loadModelFromFirebase();
+    final modelFile = await loadModelFromFirebase();
     return await loadTFLiteModel(modelFile);
   }
 
+  // Download custom model from the Firebase console and return its file
```

Comment:
nit: https://dart.dev/guides/language/effective-dart/documentation#do-use--doc-comments-to-document-members-and-types